pkg/image/controller/trigger/cache.go

```diff
@@ -0,0 +1,91 @@
+package trigger
+
+import (
+	"fmt"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	"github.com/openshift/origin/pkg/image/trigger"
+)
+
+// NewTriggerCache constructs a cacher that expects objects of type *trigger.CacheEntry
+// and converts those triggers into entries in the thread safe cache by image stream namespace
+// and name.
+func NewTriggerCache() cache.ThreadSafeStore {
+	return cache.NewThreadSafeStore(
+		cache.Indexers{
+			"images": triggerCacheIndexer,
+		},
+		cache.Indices{},
+	)
+}
+
+// triggerCacheIndexer converts a trigger cache entry into a set of image stream keys.
+func triggerCacheIndexer(obj interface{}) ([]string, error) {
+	entry := obj.(*trigger.CacheEntry)
+	var keys []string
+	for _, t := range entry.Triggers {
+		if t.From.Kind != "ImageStreamTag" || len(t.From.APIVersion) != 0 || t.Paused {
```

Comment:
Lacking a way to use it effectively, disable it by default for now.  We'd have to check groups properly and I'm hesitant to specify that here.